testhelper/smbclient.py

```
@@ -0,0 +1,82 @@
+from smb.SMBConnection import SMBConnection  # type: ignore
+import smb  # type: ignore
+import typing
+import io
+
+
+class SMBClient:
+    """Use pysmb to access the SMB server"""
+
+    def __init__(self, hostname: str, share: str, username: str, passwd: str):
+        self.server = hostname
+        self.share = share
+        self.username = username
+        self.password = passwd
+        self.error = False
+        self.connected = False
+        self.connect()
+
+    def connect(self) -> None:
+        try:
+            self.ctx = SMBConnection(
+                self.username,
+                self.password,
+                "smbclient",
+                self.server,
+                use_ntlm_v2=True,
+            )
+            self.ctx.connect(self.server)
+            self.connected = True
+        except smb.base.SMBTimeout:
+            self.error = False
```

This is confusing; I suspect most Python reader would want some exception to be raise instead of any other error-handling mechanism.

---

yeah. The SMBClient.error was used for reporting purposes on the loading test. However I agree, it is better to raise these exceptions and let the consumer handle it. One of the issue was I didn't want to raise smb.* exceptions which means the caller would then need to import the smb module. I now catch these module specific exceptions and raise a generic IOError instead.